cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/UpgradeHttpHandler.java

```diff
@@ -47,6 +48,6 @@
   @GET
   @Path("/status")
   public void getUpgradeStatus(HttpRequest request, HttpResponder responder) throws Exception {
-    responder.sendJson(HttpResponseStatus.OK, appVersionUpgradeService.getUpgradeStatus());
+    responder.sendJson(HttpResponseStatus.OK, new Gson().toJson(appVersionUpgradeService.getUpgradeStatus()));
```

Comment:
most places, the convention is to not create a new Gson object each usage, right? Pull it out to field-level object?

Reply:
fixed